### PR TITLE
feat(#261): add new folder with deployment commands and delete deploy…

### PR DESCRIPTION
**Pull request by juandiegocv27**
This is the issue #261.

Just modified and add some features for Slot CLI documentation that will bring great context with:

-deleted deployments page.
-added create, delete, update, describe, list, logs and help command pages.

image

@kariy.

### CODE REVIEW

**Comment by juandiegocv27** [comment]
Hi, @kariy I have an error, in this part, could you check it, I'll fix it as soon as possible

**Review by kariy — COMMENTED** [review]
hey @juandiegocv27 thanks for the work!

can you also update the sidebar [config](https://github.com/dojoengine/book/blob/76bc33372761cd40d4121b7facd6c1373b3d93f0/vocs.config.ts#L188) so that the new pages would be accessible from the sidebar ?

Screenshot 2024-05-13 at 9 12 24 AM

**Comment by juandiegocv27** [comment]
> hey @juandiegocv27 thanks for the work!
> 
> 
> 
> can you also update the sidebar [config](https://github.com/dojoengine/book/blob/76bc33372761cd40d4121b7facd6c1373b3d93f0/vocs.config.ts#L188) so that the new pages would be accessible from the sidebar ?
> 
> 
> 
> Screenshot 2024-05-13 at 9 12 24 AM
> 
> 

Ok sure 🫡

**Comment by juandiegocv27** [comment]
Hi @kariy, I just fix it, and agreggate some features, hope for feedback

**Comment by juandiegocv27** [comment]
HI @kariy @milancermak @milancermak, is this PR is still relevant? Please let me know if there's anything else you’d like me to add or if you have any feedback. I'm happy to make any changes you need. Thanks

**Comment by ponderingdemocritus** [comment]
@juandiegocv27 could you fix the conflict

**Comment by juandiegocv27** [comment]
@ponderingdemocritus sure, but, talking in discord with @kariy told me to erase the file deployments.md, so I could return it, but all that I do was put more stuff and make deployments a folder

**Comment by juandiegocv27** [comment]
@ponderingdemocritus @kariy just fix the conflict

**Comment by juandiegocv27** [comment]
@ponderingdemocritus @kariy just fix the conflict, but new conflict appear, now the conflict dont tell me what to fix

**Comment by juandiegocv27** [comment]
@ponderingdemocritus @kariy @milancermak @ptisserand, just resolved conflicts

**Review by kariy — CHANGES_REQUESTED** [review]

**Comment by juandiegocv27** [comment]
@ponderingdemocritus @kariy @milancermak @ptisserand Fix all the code and ready to merge: 
image

waiting for more feedback 😎🚀

File: docs/pages/toolchain/slot/Deployments/delete.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+- **Katana**: Removes a deployed Katana service.
+- **Madara**: Deletes a Madara service deployment.
+- **Torii**: Terminates a Torii service instance.
+
+### **Usage**
+
+```sh
+slot deployments delete <Project Name> <Service>
+```
```
[the remark this answers is not in the document]
Comment (juandiegocv27) [reply docs/pages/toolchain/slot/Deployments/delete.md]: this is ready

File: docs/pages/toolchain/slot/Deployments/describe.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        - **`-fork-rpc-url <URL>`**: Specify the RPC URL to fork.
+        - **`-fork-block-number <number>`**: Block number to fork from.
+        - **`-disable-fee <boolean>`**: Enable or disable fee processing.
+        - **`-gas-price <price>`**: Set the gas price.
+        - **`-invoke-max-steps <steps>`**: Maximum steps for invoke operations.
+        - **`-validate-max-steps <steps>`**: Maximum steps for validate operations.
+    - **Torii**:
+        - **`-world <world_id>`**: Identifier for the virtual world.
+        - **`-start-block <block>`**: Start block for the deployment.
+        - **`-index-pending <boolean>`**: Enable or disable indexing of pending transactions.
```
[the remark this answers is not in the document]
Comment (juandiegocv27) [reply docs/pages/toolchain/slot/Deployments/describe.md]: already

File: docs/pages/toolchain/slot/Deployments/help.md
[the remark this answers is not in the document]
Comment (juandiegocv27) [reply docs/pages/toolchain/slot/Deployments/help.md:1]: ok

File: docs/pages/toolchain/slot/Deployments/delete.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# delete
+
+The **`delete`** command is used to remove existing deployments within the Slot toolchain. This command ensures that specific services, such as Katana, Torii, or Madara, are cleanly and completely removed, along with their configurations and data.
+
+### **Purpose**
+
+This command is crucial for managing the lifecycle of services by allowing users to terminate and clean up deployments when they are no longer needed or require replacement.
```
[the remark this answers is not in the document]
Comment (juandiegocv27) [reply docs/pages/toolchain/slot/Deployments/delete.md]: I just put it some kind like that, tell me what you about it

**Review by glihm — CHANGES_REQUESTED** [review]
Thanks for the work here @juandiegocv27!

Some important points:
1. consistency: `slot` should be formatted with consistency as sometimes you use `Slot` and sometimes Slot.
2. formatting: I've left some commands on formatting, not on all the lines that require changes, you may do a general review of formatting.

**Comment by juandiegocv27** [comment]
> Thanks for the work here @juandiegocv27!
> 
> 
> 
> Some important points:
> 
> 1. consistency: `slot` should be formatted with consistency as sometimes you use `Slot` and sometimes Slot.
> 
> 2. formatting: I've left some commands on formatting, not on all the lines that require changes, you may do a general review of formatting.

@glihm thanks a lot for the feedback, I'll work on it as fast as possible ⚡️

File: docs/pages/toolchain/slot/index.md [hunk 4]
```diff
@@ -1,6 +1,12 @@
 # Slot
 
-Slot is a toolchain developed by [Cartrige.gg](https://github.com/cartridge-gg/slot) for rapidly spinning up Katana and Torii instances. Play test your game in seconds.
+Slot is a robust toolchain designed by [Cartrige.gg](https://github.com/cartridge-gg/slot) for rapidly spinning up Katana and Torii instances. which are essential for testing and development in gaming projects.
```
[the remark this answers is not in the document]
Comment (juandiegocv27) [reply docs/pages/toolchain/slot/index.md:3]: thanks for the feedback, already making changes

**Comment by juandiegocv27** [comment]
hI @glihm, @kariy, fix somethings in this issue, ready to merge🚀

File: docs/pages/toolchain/slot/deployment-commands/create.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+```sh
+slot deployments create <Project Name> <Service> [options]
+```
+
+### **Examples**
+
+1. **Create a Katana Deployment:**
+    
+    ```sh
+    slot deployments create 'MyProject' katana --version "1.0" --block-time 10
```
Comment (glihm) [inline docs/pages/toolchain/slot/deployment-commands/create.md:16]: We can remove the '' if the project has not spaces in it.
Same for the version. We can also update for a more recent version that exists:
```
slot d create example_project katana --version 1.0.0-alpha.1
```

File: docs/pages/toolchain/slot/deployment-commands/create.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    ```
+    
+2. **Initialize a Torii Service with a World:**
+    
+    ```sh
+    slot deployments create 'MyProject' torii --world 0x3fa481f41522b90b3684ecfab7650c259a76387fab9c380b7a959e3d4ac69f
```
Comment (glihm) [inline docs/pages/toolchain/slot/deployment-commands/create.md:22]: Same here to be consistent, without the ''.

File: vocs.config.ts [hunk 7]
```diff
@@ -262,11 +262,43 @@ export default defineConfig({
...
             {
-              text: "Deployments",
-              link: "/toolchain/slot/deployments-commands/deployments",
-            },
+	            text:"Reference",
+	            link: "/toolchain/slot/reference",
+	            collapsed:true,
+	            items: [
+		            {
+		              text: "deployments",
```
Comment (glihm) [inline vocs.config.ts:271]:
```suggestion
		              text: "Deployments",
```

File: docs/pages/toolchain/slot/deployment-commands/delete.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
+# delete
+
+The **`delete`** command is used to remove existing deployments within Slot. This command ensures that specific services, such as Katana, Torii, or Madara, are cleanly and completely removed, along with their configurations and data.
+s
```
Comment (glihm) [inline docs/pages/toolchain/slot/deployment-commands/delete.md:4]:
```suggestion

```

File: docs/pages/toolchain/slot/deployment-commands/delete.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+```sh
+slot deployments delete <Project Name> <Service>
+```
+
+### **Examples**
+
+1. **Delete a Katana Deployment:**
+    
+    ```sh
+    slot deployments delete 'MyProject' Katana
```
Comment (glihm) [inline docs/pages/toolchain/slot/deployment-commands/delete.md:16]: Same here let's show and example with snake case without `''`.

File: docs/pages/toolchain/slot/deployment-commands/update.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+```sh
+slot deployments update <Project Name> [service] [options]
+```
+
+### **Examples**
+
+1. **Update a Katana Deployment:**
+    
+    ```sh
+    slot deployments update 'MyProject' Katana --version "1.2" --block-time 12
```
Comment (glihm) [inline docs/pages/toolchain/slot/deployment-commands/update.md:16]: Same comment for `''` and version. 👍

File: docs/pages/toolchain/slot/deployment-commands/describe.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+```sh
+slot deployments describe <Project Name> <Service> [options]
+```
+
+### Examples
+
+1. **Describe a Katana Deployment:**
+    
+    ```sh
+    slot deployments describe 'MyProject' Katana --version "1.2"
```
Comment (glihm) [inline docs/pages/toolchain/slot/deployment-commands/describe.md:17]: Same comment for syntax, and we may actually add an output to show what's expected.

File: docs/pages/toolchain/slot/deployment-commands/logs.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    ```
+    
+2. **Stream logs from a Torii service:**
+    
+    ```sh
+    slot deployments logs 'MyProject' Torii --follow
+    ```
+    
+
+### Response and Output
```
Comment (glihm) [inline docs/pages/toolchain/slot/deployment-commands/logs.md:27]: Those sections with response and output are great. Could we generalize it to the other commands showing real output + comments on it when required?

File: docs/pages/toolchain/slot/deployment-commands/logs.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
+# logs
+
+The **`logs`** command is used to fetch and display logs from a specified deployment service within the Slot environment. This command supports real-time log streaming as well as querying historical logs, making it an essential tool for monitoring and debugging deployment activities.
+
+### Usage
```
Comment (glihm) [inline docs/pages/toolchain/slot/deployment-commands/logs.md:5]: Comment that applies for all, let's use `##` here as the first level if `#`.

Like so, if we want to insert a new subsection, we can use `###`.

**Review by glihm — COMMENTED** [review]
@juandiegocv27 thanks for the work here! Some comments that also apply to all commands.

Once done, this should be ready for merging!
Thank you for the iterations, let's know if you have any question!

**Comment by juandiegocv27** [comment]
> @juandiegocv27 thanks for the work here! Some comments that also apply to all commands.
> 
> Once done, this should be ready for merging! Thank you for the iterations, let's know if you have any question!

Thank you for the feedback @glihm, I´ll make the changes as fast as possible

**Comment by glihm** [comment]
> > @juandiegocv27 thanks for the work here! Some comments that also apply to all commands.
> > Once done, this should be ready for merging! Thank you for the iterations, let's know if you have any question!
> 
> Thank you for the feedback @glihm, I´ll make the changes as fast as possible

No rush, thank you for your contribution. 👍

**Comment by juandiegocv27** [comment]
> > > @juandiegocv27 thanks for the work here! Some comments that also apply to all commands.
> > > Once done, this should be ready for merging! Thank you for the iterations, let's know if you have any question!
> > 
> > 
> > Thank you for the feedback @glihm, I´ll make the changes as fast as possible
> 
> No rush, thank you for your contribution. 👍

Already fix and put response and output in every command😎 @glihm

**Review by kariy — APPROVED** [review]
lgtm

**Comment by juandiegocv27** [comment]
> lgtm

thank you so much

**Pull request merged**